pkg/comment/comment_message_loader.go

```diff
@@ -0,0 +1,104 @@
+package comment
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"regexp"
+	"text/template"
+
+	assets "github.com/arquillian/ike-prow-plugins/pkg/assets/generated"
+	"github.com/arquillian/ike-prow-plugins/pkg/config"
+	"github.com/arquillian/ike-prow-plugins/pkg/github/service"
+	"github.com/arquillian/ike-prow-plugins/pkg/scm"
+	"github.com/arquillian/ike-prow-plugins/pkg/utils"
+)
+
+// Message struct for the plugin comment service
+type Message struct {
+	Thumbnail      string
+	Description    string
+	ConfigFile     string
+	Documentation  string
+	MessageFileURL string
+}
+
+var commentMessage Message
+
+// LoadMessageTemplate loads a comment message for the plugins from the template files. If the comment message is set in config then it takes that one, the default otherwise.
+func LoadMessageTemplate(message Message, configuration config.PluginConfiguration, change scm.RepositoryChange) string {
+	var msg string
+	commentMessage = message
+	if configuration.PluginHint != "" {
+		msg = getMsgFromConfigHint(configuration, change)
+	} else if content := defaultFileContent(configuration, change); content != "" {
+		msg = content
+	} else if configuration.LocationURL == "" {
+		msg = loadMessageTemplate("message-with-no-config.txt")
+	} else {
+		msg = loadMessageTemplate("message-with-config.txt")
+	}
+	return getMsgFromTemplate(msg)
+}
+
+func getMsgFromTemplate(msg string) string {
+	var tpl bytes.Buffer
+	msgTemplate := template.Must(template.New("message").Parse(msg))
```

Comment:
Please, avoid using `*must*` methods that fail with a panic - then, we have no way to easily get the reason of the failure - it is not sent to the Sentry. In addition `template.Must` does nothing else than just checking the error and failing with panic if error is not null:
```go
func Must(t *Template, err error) *Template {
	if err != nil {
		panic(err)
	}
	return t
}
```